src/Core/Domain/CartRule/QueryResult/EditableCartRuleConditions.php

```diff
@@ -69,7 +69,7 @@ class EditableCartRuleConditions
     private $restrictions;
 
     public function __construct(
-        ?CustomerId $customerId,
+        ?CustomerIdInterface $customerId,
```

Comment:
From the only use case where this constructor is used it seems like `$customerId` is always provided so I don't think the input here should be nullable

Reply:
```suggestion
        CustomerIdInterface $customerId,
```